src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -142,40 +156,40 @@
     (comidi/GET ["/resource_types/" [#".*" :rest]] request
       (master-request-handler request))
     (comidi/GET ["/status/" [#".*" :rest]] request
-      (master-request-handler request))
+      (master-request-handler request))))
 
-    ;; legacy CA routes
+(defn legacy-ca-routes
+  [ca-request-handler]
+  (comidi/routes
     (comidi/ANY ["/certificate_status/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/ANY ["/certificate_statuses/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate_revocation_list/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate_request/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/PUT ["/certificate_request/" [#".*" :rest]] request
-      (ca-request-handler request))))
-
-(defn legacy-routes
-  "Creates all of the comidi routes for the master."
-  [master-request-handler ca-request-handler]
-  (comidi/routes
-    (comidi/context "/v2.0"
-      (v2_0-routes master-request-handler))
-    (comidi/context ["/" :environment]
-      (legacy-master-routes master-request-handler ca-request-handler))))
+                (ca-request-handler request))))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
-(defn build-ring-handler
-  [master-handler master-mount master-api-version
-   ca-handler ca-mount ca-api-version]
-  (comidi/routes->handler
-    (legacy-routes
-      #(request-compatibility-wrapper
-        master-handler master-mount master-api-version %)
-      #(request-compatibility-wrapper
-        ca-handler ca-mount ca-api-version %))))
+(schema/defn ^:always-validate
+  build-ring-handler :- IFn
```

Comment:
FWIW, I really hate using indentation everywhere. I only care that we avoid unnecessary whitespace changes in commits that are focused on something else, and in most cases that really only means let bindings and maps. I am very opinionated that the values of those shouldn't be vertically aligned. Otherwise, I really don't care that much, so long as I can make my editor do whatever we've decided on. I dislike how indenting things in e.g. a thread macro looks, but my editor can do it, so whatever.